core/src/main/java/com/linecorp/armeria/client/cookie/CookieJar.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.cookie;
+
+import java.net.URI;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.Cookie;
+import com.linecorp.armeria.common.Cookies;
+
+import io.netty.util.NetUtil;
+
+/**
+ * A {@link Cookie} container for a client.
+ */
+public interface CookieJar {
+
+    /**
+     * Determines whether a host matches a domain, as specified by RFC 6265
+     * <a href="https://tools.ietf.org/html/rfc6265#section-5.1.3">Domain Matching</a>.
+     */
+    static boolean domainMatch(@Nullable String domain, @Nullable String host) {
```

Comment:
How about moving this to `CookiePolicy` as a member method instead of a static one, so that a user can implement one's own domain matching logic when implementing a `CookiePolicy`? i.e.

```java
public interface CookiePolicy {
    ...
    default boolean domainMatches(@Nullable String doman, @Nullable String host) {
        // .. our default implementation goes here ..
    }
}
```

Reply:
Otherwise, the domain matching specs are standardized and specified in RFC 6265. 
Could we remove this method from public API and move to implementation detail?